controller/appcontroller.go

```diff
@@ -617,7 +617,7 @@ func (ctrl *ApplicationController) getAppHosts(a *appv1.Application, appNodes []
 		sort.Slice(resourcesInfo, func(i, j int) bool {
 			return resourcesInfo[i].ResourceName < resourcesInfo[j].ResourceName
 		})
-		hosts = append(hosts, appv1.HostInfo{Name: nodeName, SystemInfo: node.SystemInfo, ResourcesInfo: resourcesInfo})
+		hosts = append(hosts, appv1.HostInfo{Name: nodeName, SystemInfo: node.SystemInfo, ResourcesInfo: resourcesInfo, Labels: node.Labels}) //TODO: Only append labels defined in conf
```

Comment:
My spontaneous thought is that this could be managed on an `AppProject` level, i.e. to have a whitelist of node labels as an attribute of an `AppProject`.

Something like

```yaml
allowedNodeLabels:
   - kubernetes.io/arch
   - kubernetes.io/os
   - topology.kubernetes.io/zone
   - node.kubernetes.io/instance-type
   - karpenter.k8s.aws/instance-cpu
   - karpenter.k8s.aws/instance-memory
```

WDYT @crenshaw-dev?